pkg/sql/plan/build_show.go

```diff
@@ -202,31 +205,31 @@ func buildShowCreateView(stmt *tree.ShowCreateView, ctx CompilerContext) (*Plan,
 
 	// logutil.Info(sqlStr)
 
-	return returnByRewriteSQL(ctx, sqlStr, plan.DataDefinition_SHOW_CREATETABLE)
+	return returnByRewriteSQL(ctx, sqlStr, plan.DataDefinition_SHOW_CREATETABLE, accountId)
 }
 
-func buildShowDatabases(stmt *tree.ShowDatabases, ctx CompilerContext) (*Plan, error) {
+func buildShowDatabases(stmt *tree.ShowDatabases, ctx CompilerContext, accountId uint32) (*Plan, error) {
 	if stmt.Like != nil && stmt.Where != nil {
 		return nil, moerr.NewSyntaxError("like clause and where clause cannot exist at the same time")
 	}
 	ddlType := plan.DataDefinition_SHOW_DATABASES
-	sql := fmt.Sprintf("SELECT datname `Database` FROM %s.mo_database", MO_CATALOG_DB_NAME)
+	sql := fmt.Sprintf("SELECT datname `Database` FROM %s.mo_database where account_id = %v or account_id = 0", MO_CATALOG_DB_NAME, accountId)
```

Comment:
Why do you need to pass account id around?   Can you just get account id here, so that there is no need to add param everywhere.     I assume we can still get the context at this moment.

Reply:
> Why do you need to pass account id around? Can you just get account id here, so that there is no need to add param everywhere. I assume we can still get the context at this moment.

Context not passed. Very embarrassing.

Reply:
Anyway, need to add a parameter, either pass the context or the accountid. i chose the latter. @fengttt

Reply:
Another approach is to add a GetAccountID() method to the CompilerContext, and then modify the frontend structure to add accountID or context information to it, so that it can return to implement the GetAccountID() method

Reply:
Account id context should be part of the compiler context.

Reply:
> Account id context should be part of the compiler context.

ok